diff --git a/src/voc.c b/src/voc.c
index a75639e9..188d11bf 100644
--- a/src/voc.c
+++ b/src/voc.c
@@ -344,6 +344,11 @@ static size_t read_samples(sox_format_t * ft, sox_sample_t * buf,
           break;
       }
 
+      if (v->size < 2) {
+	lsx_warn("VOC input: short file");
+	return 0;
+      }
+
       /* Read the data in the file */
       if (v->size <= 4) {
         if (!v->adpcm.setup.sign) {
