diff --git a/src/voc.c b/src/voc.c
index a75639e9..4564afb1 100644
--- a/src/voc.c
+++ b/src/voc.c
@@ -327,6 +327,11 @@ static size_t read_samples(sox_format_t * ft, sox_sample_t * buf,
   if (v->block_remaining == 0)  /* if no more data, return 0, i.e., done */
     return 0;
 
+  if (v->size == 0) {
+    lsx_warn("VOC input: zero file size");
+    return 0;
+  }
+
   if (v->silent) {
     for (; v->block_remaining && (done < len); v->block_remaining--, done++)
       *buf++ = 0;       /* Fill in silence */
