From 507b669990d051546050eb912c9fd75fe22e97fc Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Thu, 23 May 2024 20:03:03 +0100
Subject: [PATCH] spectrogram: fix crash if pixels per second exceeds sample
 rate

---
 src/spectrogram.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/spectrogram.c b/src/spectrogram.c
index 3dcda69c..89ae1fbf 100644
--- a/src/spectrogram.c
+++ b/src/spectrogram.c
@@ -431,7 +431,7 @@ static int start(sox_effect_t *effp)
   actual = make_window(p, p->last_end = 0);
   lsx_debug("window_density=%g", actual / p->dft_size);
   p->step_size = (p->slack_overlap ? sqrt(actual * p->dft_size) : actual) + 0.5;
-  p->block_steps = effp->in_signal.rate / pixels_per_sec;
+  p->block_steps = max(effp->in_signal.rate / pixels_per_sec, 1);
   p->step_size =
     p->block_steps / ceil((double)p->block_steps / p->step_size) + 0.5;
   p->block_steps = floor((double)p->block_steps / p->step_size + 0.5);
-- 
2.39.2

